scripts/docker-dos/dosUrlLocalizer.sc

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env amm
+
+interp.repositories() ++= Seq(
+  coursier.maven.MavenRepository("https://oss.sonatype.org/content/repositories/snapshots/"))
+
+@
+import $ivy.`org.http4s::http4s-dsl:0.19.0-SNAPSHOT`
+import $ivy.`org.http4s::http4s-blaze-server:0.19.0-SNAPSHOT`
+import $ivy.`org.http4s::http4s-blaze-client:0.19.0-SNAPSHOT`
+import $ivy.`com.google.cloud:google-cloud-storage:1.35.0`
+import $ivy.`com.google.oauth-client:google-oauth-client:1.23.0`
+import $ivy.`com.google.auth:google-auth-library-credentials:0.9.1`
+import $ivy.`org.http4s::http4s-circe:0.19.0-SNAPSHOT`
+import $ivy.`io.circe::circe-literal:0.7.0`
+import $ivy.`io.spray::spray-json:1.3.4`
+
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.cloud.storage.Storage
+import com.google.cloud.storage.StorageOptions
+import com.google.common.collect.Lists
+import com.google.cloud.ReadChannel
+import java.io.ByteArrayInputStream
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import org.http4s.client.blaze._
+import cats.effect._
+import org.http4s.dsl.io._
+import org.http4s._
+import org.http4s.circe._
+import io.circe.literal._
+import io.circe.syntax._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import scala.util.{Try, Success, Failure}
+
+object MarthaResponseJsonSupport extends DefaultJsonProtocol {
+  implicit val urlFormat: JsonFormat[Url] = jsonFormat1(Url)
+  implicit val dataObject: JsonFormat[DosDataObject] = jsonFormat1(DosDataObject)
+  implicit val dosObjectFormat: JsonFormat[DosObject] = jsonFormat1(DosObject)
+  implicit val marthaResponseFormat: JsonFormat[MarthaResponse] = jsonFormat2(MarthaResponse)
+}
+
+case class Url(url: String)
+
+case class DosDataObject(urls: Array[Url])
+
+case class DosObject(data_object: DosDataObject)
+
+case class MarthaResponse(dos: DosObject, googleServiceAccount: JsObject)
+
+
+@main
+def dosUrlResolver(dosUrl: String, downloadLoc: String) : Unit = {
+  val dosResloverObj = for {
+    marthaResObj <- resolveDosThroughMartha(dosUrl)
+    _ <- downloadFileFromGcs(marthaResObj.dos.data_object.urls(0).url, marthaResObj.googleServiceAccount.toString, downloadLoc)
```

Comment:
find the first instance of a url starting with `gs://` and throw an exception if none found

Reply:
👍